### PR TITLE
VSS API

### PR DESCRIPTION
**Pull request by alonre24**
A draft PR to mark the comments in the code that is about to be merged to master. 
Fixes as result of the comments will be merged into feature_vecsim from another branch.

### CODE REVIEW

**Review by ashtul — COMMENTED** [review]
mostly small comments

File: src/hybrid_reader.c [hunk 1]
```diff
@@ -0,0 +1,327 @@
...
+  hi->iter = NULL;
+
+  if (child_it == NULL) {
+    hi->mode = STANDARD_KNN;
+//  } else if (UseBF(child_it->NumEstimated(child_it), query, index)) {
+//    hi->mode = HYBRID_ADHOC_BF;
+  } else {
+    hi->mode = HYBRID_BATCHES;
+    //Todo: apply heuristics (batch_size = k / (vec_index_size*child_it->NumEstimated(child_it)))
+  }
```
[the remark this answers is not in the document]
Comment (alonre24) [reply src/hybrid_reader.c:298]: That's the plan, will be implemented soon... :)

File: src/ext/default.c [hunk 2]
```diff
@@ -146,7 +146,9 @@ static double bm25Recursive(const ScoringFunctionArgs *ctx, const RSIndexResult
   static const float k1 = 1.2;
   double f = (double)r->freq;
   double ret = 0;
-
+  if (r->type == RSResultType_HybridDistance) {
+    RS_LOG_ASSERT(false, "Scoring function is not supported with vector similarity search")
```
[the remark this answers is not in the document]
Comment (alonre24) [reply src/ext/default.c:150]: I added support for this scoring

File: src/ext/default.c [hunk 3]
```diff
@@ -226,6 +228,8 @@ static double dismaxRecursive(const ScoringFunctionArgs *ctx, const RSIndexResul
   // for terms - we return the term frequency
   double ret = 0;
   switch (r->type) {
+    case RSResultType_HybridDistance:
+      RS_LOG_ASSERT(false, "Scoring function is not supported with vector similarity search")
```
[the remark this answers is not in the document]
Comment (alonre24) [reply src/ext/default.c:232]: I added support for this scoring

**Review by gkorland — APPROVED** [review]

**Review by rafie — APPROVED** [review]

**Pull request merged**